### PR TITLE
Client connection timeout

### PR DESCRIPTION
**Pull request by nildot**
Clean unused code for com.vesoft.nebula.Client

Add connection timeout for Clients. The default timeout for TSocket is 0, which will block client infinitely when server behave strangely

### CODE REVIEW

File: client/src/main/java/com/vesoft/nebula/Client.java [hunk 1]
```diff
@@ -14,7 +14,7 @@
  */
 public interface Client extends AutoCloseable, Serializable {
 
-    public static enum NebulaCode {
```
Comment (darionyaphet) [inline client/src/main/java/com/vesoft/nebula/Client.java:17]: why  remove it?

**Comment by darionyaphet** [comment]
Oh it's `Socket connection timeout`

File: client/src/main/java/com/vesoft/nebula/AbstractClient.java [hunk 2]
```diff
@@ -33,9 +35,10 @@
      * @param connectionRetry The number of retries when connection failure.
      * @param executionRetry  The number of retries when execution failure.
      */
-    public AbstractClient(List<HostAndPort> addresses, int timeout,
+    public AbstractClient(List<HostAndPort> addresses, int timeout, int connTimeout,
                           int connectionRetry, int executionRetry) {
         checkArgument(timeout > 0);
+        checkArgument(connTimeout >= 0);
```
Comment (darionyaphet) [inline client/src/main/java/com/vesoft/nebula/AbstractClient.java:41]: the `connTimeout` could be zero?

File: client/src/main/java/com/vesoft/nebula/Client.java (hunk 1 above)
Comment (nildot) [reply client/src/main/java/com/vesoft/nebula/Client.java:17]: Enum is static and public by default

File: client/src/main/java/com/vesoft/nebula/AbstractClient.java (hunk 2 above)
Comment (nildot) [reply client/src/main/java/com/vesoft/nebula/AbstractClient.java:41]: See the constructor of TSocket, connectionTimeout is initialized with 0 by default

Comment (darionyaphet) [reply client/src/main/java/com/vesoft/nebula/AbstractClient.java:41]: Do you mind using `connectionTimeout` ?

File: client/src/main/java/com/vesoft/nebula/Client.java (hunk 1 above)
Comment (darionyaphet) [reply client/src/main/java/com/vesoft/nebula/Client.java:17]: The interface's method should be public. So we should specified it explicitly

**Comment by darionyaphet** [comment]
Generally LGTM 👍

File: client/src/main/java/com/vesoft/nebula/Client.java (hunk 1 above)
Comment (nildot) [reply client/src/main/java/com/vesoft/nebula/Client.java:17]: Methods defined in an interface is public as default. Constant values are public and static as default
[The Java™ Tutorials](https://docs.oracle.com/javase/tutorial/java/IandI/interfaceDef.html#:~:text=The%20Interface%20Body,-The%20interface%20body&text=All%20abstract%2C%20default%2C%20and%20static,public%20%2C%20static%20%2C%20and%20final%20.)

File: client/src/main/java/com/vesoft/nebula/AbstractClient.java (hunk 2 above)
Comment (nildot) [reply client/src/main/java/com/vesoft/nebula/AbstractClient.java:41]: OK. connectionTimeout looks better

**Comment by nildot** [comment]
Rename `connTimeout` to `connectionTimeout`

**Pull request merged**